src/main/java/io/numaproj/numaflow/function/reduce/ReduceSupervisorActor.java

```diff
@@ -33,26 +33,26 @@
 
 @Slf4j
 public class ReduceSupervisorActor extends AbstractActor {
-    private final Class<? extends Reducer> groupBy;
+    private final ReducerFactory<? extends Reducer> reducerFactory;
```

Comment:
why don't we pass `private final Reducer reducer;` instead of the factory to the actor?

Reply:
same comment